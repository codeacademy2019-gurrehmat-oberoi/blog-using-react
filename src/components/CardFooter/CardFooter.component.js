import React from 'react';
import './CardFooter.css'

const CardFooter = (props) => {
  const claps = props.claps;
  const liked = props.liked;
  // const likedIcon = ;
  return (
    <div className='CardFooter'>
      <div className='left'>
        <img src={require('../../Icons/clapping.svg')} className = 'icon clap' alt='' onClick = {()=>props.onClapAction()}/>
        <span className='clapCount'> {claps}</span>
      </div>
      <div className='right'>
        <img src= {require(`../../Icons/heart-${ liked ? 'red':'black'}.svg`)} className = "icon heart" alt= '' onClick = {() => props.onLikeAction()}/>
      </div>
    </div>
  );
}

CardFooter.defaultProps = {
  claps:10,
  liked:false,
}

export default CardFooter;