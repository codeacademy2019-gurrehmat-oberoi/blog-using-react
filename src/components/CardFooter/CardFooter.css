.CardFooter {
  width: 100%;
  display: flex;
  flex-direction: row;
  /* justify-content: space-between; */
  align-items: center;
}

.icon {
  width:20px;
  margin: 5px;
}

.left {
  width: 15%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
}
.right {
  display: flex;
  justify-content: flex-end;
  width: 85%;
  align-items: center;
}