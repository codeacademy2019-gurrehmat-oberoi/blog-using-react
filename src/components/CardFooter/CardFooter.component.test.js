import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import CardFooter from './CardFooter.component';

describe('CardFooter', () => {
  it('should render a Card Footer', () => {
    const tree = renderer.create(<CardFooter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show red heart when prop liked is true', () => {
    const wrapper = shallow(<CardFooter liked={true} />);
    expect(wrapper.find('.heart').props().src.default).toEqual('heart-red.svg');
  });

  it('should show black heart when prop liked is false', () => {
    const wrapper = shallow(<CardFooter liked={false} />);
    expect(wrapper.find('.heart').props().src.default).toEqual('heart-black.svg');
  });
})