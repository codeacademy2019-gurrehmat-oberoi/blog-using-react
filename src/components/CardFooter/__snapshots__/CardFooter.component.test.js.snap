// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter should render a Card Footer 1`] = `
<div
  className="CardFooter"
>
  <div
    className="left"
  >
    <img
      alt=""
      className="icon clap"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "clapping.svg",
        }
      }
    />
    <span
      className="clapCount"
    >
       
      10
    </span>
  </div>
  <div
    className="right"
  >
    <img
      alt=""
      className="icon heart"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "heart-black.svg",
        }
      }
    />
  </div>
</div>
`;
